src/main/fc/fc_init.c

```diff
@@ -450,6 +451,43 @@ void init(void)
 #endif
 #endif // USE_SPI
 
+#ifdef USB_MSC
+#ifdef MSC_BUTTON
+    IO_t msc_button = IOGetByTag(IO_TAG(MSC_BUTTON));
+	IOInit(msc_button, OWNER_USB, 0);
+#ifdef MSC_BUTTON_IPU
+	IOConfigGPIO(msc_button, IOCFG_IPU);
+#define BUTTON_STATUS IORead(msc_button) == 0
+#else
+	IOConfigGPIO(msc_button, IOCFG_IPD);
+#define BUTTON_STATUS IORead(msc_button) == 1
+#endif
+#else
+#define BUTTON_STATUS 0
+#endif
+
+/* MSC mode will start after init, but will not allow scheduler to run,
+ *  so there is no bottleneck in reading and writing data */
+    if (*((uint32_t *)0x2001FFF0) == 0xDDDD1010 || BUTTON_STATUS) {
```

Comment:
This should probably go into a file of its own, or at least a function of its own.

Reply:
Also, this is a mix of spaces and tabs. Can you please change this to use 4 spaces for indentation?